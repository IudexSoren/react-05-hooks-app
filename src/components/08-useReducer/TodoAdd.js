import { useForm } from '../../hooks/useForm';

const TodoAdd = ({ handleAddTodo }) => {

  const [{ description }, handleInputChange, reset] = useForm({
    description: ''
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length <= 1) {
      return;
    }
    const newTodo = {
      id: new Date().getTime(),
      desc: description,
      done: false
    }
    handleAddTodo(newTodo);
    reset();
  }

  return (
    <>
      <h4>Agregar ToDo</h4>
      <hr/>
      <form onSubmit={ handleSubmit }>
        <input
          type="text"
          className="form-control mb-3"
          name="description"
          value={ description }
          onChange={ handleInputChange }
          placeholder="DescripciÃ³n"
          autoComplete="off"
        />
        <div className="d-grid gap-2">
          <button
            className="btn btn-outline-success"
            type="submit"
          >Agregar</button>
        </div>
      </form>
    </>
  )
}

export default TodoAdd
