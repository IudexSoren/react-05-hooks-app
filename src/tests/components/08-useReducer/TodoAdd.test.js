import { shallow } from 'enzyme';
import TodoAdd from '../../../components/08-useReducer/TodoAdd';

describe('Pruebas de TodoAdd', () => {

  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={ handleAddTodo } />);

  test('Debe renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('No debe llamar la función handleAddTodo', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(handleAddTodo).not.toHaveBeenCalled();
  });

  test('Debe llamar la función handleAddTodo', () => {
    const value = 'Aprender MongoDB';
    wrapper.find('input').simulate('change', {
      target: {
        name: 'description',
        value: value
      }
    });
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(handleAddTodo).toHaveBeenCalled();
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false
    });
    expect(wrapper.find('input').prop('value')).toBe('');
  });

});
