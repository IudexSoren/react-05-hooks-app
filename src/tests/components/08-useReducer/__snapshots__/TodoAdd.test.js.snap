// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de TodoAdd Debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h4>
    Agregar ToDo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control mb-3"
      name="description"
      onChange={[Function]}
      placeholder="DescripciÃ³n"
      type="text"
      value=""
    />
    <div
      className="d-grid gap-2"
    >
      <button
        className="btn btn-outline-success"
        type="submit"
      >
        Agregar
      </button>
    </div>
  </form>
</Fragment>
`;
